### ISSUE TITLE
Cannot find module based on relative path

### ISSUE THREAD

**Issue by kukko** [thlorenz/proxyquire#250]
Hello!

I wanted to use proxyquire in unit tests in a project, but when I try to mock a not existing module for the tests I constantly get the following error:
Error: Cannot find module './TestService/TestService.js'

Below is the code for the module I want to test and the test where I want to use proxyquire. The module for what I want to make a stub, does not exists in the file system. Is it possible to make a stub for modules which is not exists? And I have to provide the relative path for the faked module from the test, to that path where the tested module will search it?

Here is the code what I want to test ('./src/services/ServiceProvider.js'):
```
class ServiceProvider{
    static getRouteService(){
        return this.getService('RouteService');
    }
    static getService(serviceType){
        if (typeof this.services[serviceType] === "undefined"){
            this.loadService(serviceType);
        }
        return this.services[serviceType];
    }
    static loadService(serviceType){
        let { join, resolve, relative } = require("path"),
            servicePath = "./" + serviceType + "/" + serviceType + ".js";
        this.services[serviceType] = require(servicePath);
    }
    static clearCache(){
        this.services = {};
    }
}

ServiceProvider.services = {};

module.exports = ServiceProvider;
```

And here is the test ('test/unitTests/serviceProvider.js') for the module above:
```
let { assert } = require('chai'),
    sinon = require('sinon'),
    proxyquire = require('proxyquire').noCallThru();

describe('ServiceProvider', () => {
    let ServiceProvider = require('../../index.js').ServiceProvider;
        /**
         * Here are som other tests but there I didn't used proxyquire.
         */
        describe('loadService', () => {
            describe('Path package', () => {
                let serviceName = 'TestService';
                before(() => {
                    let proxy = {};
                    proxy['../../src/services/TestService/TestService.js'] = {
                        name: serviceName,
                        '@noCallThru': true
                    };
                    proxyquire('../../src/services/ServiceProvider.js', proxy);
                    ServiceProvider.loadService(serviceName);
                });
                it('TestService is not undefined', () => {
                    assert.isDefined(ServiceProvider.services[serviceName]);
                });
                it('TestService is object', () => {
                    assert.isObject(ServiceProvider.services[serviceName]);
                });
            });
            after(() => {
                sinon.restore();
            });
        });
    });
});
```

And here is the directory structure for the project:
```
- src
-- services
--- Folder for each existing service. The TestService folder should be here.
--- ServiceProvider.js
- test
-- unitTests
--- serviceProvider.js
```

**Comment by bendrucker** [comment]
Stubs should be relative to the file you're trying to proxyquire, not the current script. Your problem is here:

```
proxy['../../src/services/TestService/TestService.js']
```

**Comment by kukko** [comment]
I already tried to specify the stub path relatively to the proxyquired path, but I got the same error.

So the stub's path was pecified like this:
`proxy['./TestService/TestService.js']`
And I used proxyquire like this:
`proxyquire('../../src/services/ServiceProvider.js', proxy);`

And the ServiceProvider.js searchs the TestService.js in the `./TestService/TestService.js` path.

I load the ServiceProvider module from the `../../index.js` file so I also tried to specify the stub's path relatively to the `../../index.js` file and tried to call the proxyquire method with this path: `../../index.js`

So I don't understand what can be the problem, because none of the above methods worked.
